Chatto/Source/ChatController/BaseChatViewController.swift

```diff
@@ -265,20 +273,29 @@ open class BaseChatViewController: UIViewController, UICollectionViewDataSource,
 
         self.collectionView.scrollIndicatorInsets = {
             var currentInsets = self.collectionView.scrollIndicatorInsets
-            currentInsets.bottom = self.constants.defaultScrollIndicatorInsets.bottom + inputHeightWithKeyboard
-            currentInsets.top = self.topLayoutGuide.length + self.constants.defaultScrollIndicatorInsets.top
+            currentInsets.bottom = self.layoutConfiguration.scrollIndicatorInsets.bottom + inputHeightWithKeyboard
+            currentInsets.top = self.topLayoutGuide.length + self.layoutConfiguration.scrollIndicatorInsets.top
             return currentInsets
         }()
 
-        let inputIsAtBottom = self.view.bounds.maxY - self.inputContainer.frame.maxY <= 0
+        guard shouldUpdateContentOffset else { return }
 
+        let inputIsAtBottom = self.view.bounds.maxY - self.inputContainer.frame.maxY <= 0
         if allContentFits {
-            self.collectionView.contentOffset.y = -self.collectionView.contentInset.top
+            var contentOffset = self.collectionView.contentOffset
+            contentOffset.y = -self.collectionView.contentInset.top
+            self.updateContentViewOffset(contentOffset)
         } else if !isInteracting || inputIsAtBottom {
-            self.collectionView.contentOffset.y = newContentOffsetY
+            var contentOffset = self.collectionView.contentOffset
+            contentOffset.y = newContentOffsetY
+            self.updateContentViewOffset(contentOffset)
         }
     }
 
+    open func updateContentViewOffset(_ newValue: CGPoint) {
```

Comment:
Why do you need this function? Is it the only function that should be called when you modify content offset? It's not used everywhere now, so, it's quite confusing. If you want to use it only when you adjust content insets then it's better to extract related code from `adjustContentInsets` into `adjustContentOffset` function and call it directly when needed. In this case, you won't need to pass any additional parameter to `adjustContentInsets` function